rethinkdb_delete_block.py

```diff
@@ -0,0 +1,49 @@
+import rethinkdb as rdb
+from nio.properties import StringProperty, Property
+from nio.util.discovery import discoverable
+from .rethinkdb_base_block import RethinkDBBase
+from nio.signal.base import Signal
+
+
+@discoverable
+class RethinkDBDelete(RethinkDBBase):
+
+    """a block for deleting one or more documents from a rethinkdb table"""
+
+    table = StringProperty(title="Table to query", default='test',
+                           allow_none=False)
+    filter = Property(title='Filter by given fields',
+                      default='{{ $.to_dict() }}',
+                      allow_none=False)
+
+    def _locked_process_signals(self, signals):
+        notify_list = []
+        for signal in signals:
+            self.logger.debug('Delete is Processing signal: {}'.format(signal))
+            # update incoming signals with results of the query
+            delete_results = self.execute_with_retry(self._delete, signal)
+            self.logger.debug("Delete results: {}".format(delete_results))
+            notify_list.append(Signal(delete_results))
```

Comment:
All the other blocks use the enrich signals mixin so we should use that here too.